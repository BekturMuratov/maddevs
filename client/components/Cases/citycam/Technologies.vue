<template>
  <section class="container_regular">
    <h2 class="case_title_h2 m-24_bottom media-m-12_bottom">
      Technology stack
    </h2>
    <ListTechnologies class="m-96_bottom media-m-48_bottom">
      <ListTechnologiesItem
        v-for="technology in technologies"
        :key="technology.name"
        class="case_tech-list-item"
        v-bind="technology"
      />
    </ListTechnologies>
  </section>
</template>

<script>
import ListTechnologies from '@/components/Cases/shared/ListTechnologies'
import ListTechnologiesItem from '@/components/Cases/shared/ListTechnologiesItem'
import { technologies } from '@/data/caseCityCam'

export default {
  name: 'Technologies',
  components: {
    ListTechnologies,
    ListTechnologiesItem,
  },

  data() {
    return {
      technologies,
    }
  },
}
</script>

<style lang="scss" scoped>
.case {
  &_tech-list-item::v-deep {
    p {
      font-size: 12px;
    }

    .django {
      width: 57px;
    }

    @media screen and (max-width: 768px) {
      p {
        font-size: 8px;
      }
      .tech{
        &-texas-instruments,
        &-altium-designer{
          white-space: normal;
          line-height: 7.5px;
        }
      }

      .django {
        width: 30px;
      }
    }
  }
}
</style>
