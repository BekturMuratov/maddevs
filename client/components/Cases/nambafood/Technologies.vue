<template>
  <section class="container_regular">
    <TextParagraph class="m-24_top media-m-48_top">
      Monitoring also helps Mad Devs to prevent service timeouts. When the load grows, we can track spikes, estimate the
      severity of an issue and quickly resolve it.
    </TextParagraph>
    <h3 class="case_title_h3 m-96_top m-48_bottom media-m-72_top media-m-24_bottom">
      Technologies used for the Namba Food project
    </h3>
    <ListTechnologies>
      <ListTechnologiesItem
        v-for="technology in technologies"
        :key="technology.name"
        :tech-name="technology.name"
        :class-name="technology.className"
      />
    </ListTechnologies>
  </section>
</template>

<script>
import TextParagraph from '@/components/Cases/TextParagraph'
import ListTechnologies from '@/components/Cases/ListTechnologies'
import ListTechnologiesItem from '@/components/Cases/ListTechnologiesItem'
import { technologies } from '@/data/caseNambafood'

export default {
  name: 'Technologies',
  components: {
    TextParagraph,
    ListTechnologies,
    ListTechnologiesItem,
  },

  data() {
    return {
      technologies,
    }
  },
}
</script>

<style></style>