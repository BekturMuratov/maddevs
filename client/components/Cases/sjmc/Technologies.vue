<template>
  <section class="container_regular">
    <h3 class="case_title_h3 m-48_bottom media-m-24_bottom">
      Technology stack
    </h3>
    <ListTechnologies class="m-107_bottom media-m-48_bottom">
      <ListTechnologiesItem
        v-for="technology in technologies"
        :key="technology.name"
        v-bind="technology"
      />
    </ListTechnologies>
  </section>
</template>

<script>
import ListTechnologies from '@/components/Cases/shared/ListTechnologies'
import ListTechnologiesItem from '@/components/Cases/shared/ListTechnologiesItem'
import { technologies } from '@/data/caseSJMC'

export default {
  name: 'Technologies',
  components: {
    ListTechnologies,
    ListTechnologiesItem,
  },

  data() {
    return {
      technologies,
    }
  },
}
</script>
