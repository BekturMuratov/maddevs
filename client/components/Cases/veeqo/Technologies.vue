<template>
  <section class="container_regular">
    <h2 class="case_title_h2 case_text-align-center m-24_bottom media-m-12_bottom">
      Technology stack
    </h2>
    <ListTechnologies class="m-96_bottom media-m-48_bottom">
      <ListTechnologiesItem
        v-for="technology in technologies"
        :key="technology.name"
        v-bind="technology"
      />
    </ListTechnologies>
  </section>
</template>

<script>
import ListTechnologies from '@/components/Cases/shared/ListTechnologies'
import ListTechnologiesItem from '@/components/Cases/shared/ListTechnologiesItem'
import { technologies } from '@/data/caseVeeqo'

export default {
  name: 'Technologies',
  components: {
    ListTechnologies,
    ListTechnologiesItem,
  },

  data() {
    return {
      technologies,
    }
  },
}
</script>
