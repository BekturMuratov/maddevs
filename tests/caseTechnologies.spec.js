import { mount } from '@vue/test-utils'
import Technologies from '../client/components/Cases/yourcast/Technologies'

describe('technology', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(Technologies)
  })

  // ------ IMPORTANT ----- //
  it('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  // --------------------- //
})
