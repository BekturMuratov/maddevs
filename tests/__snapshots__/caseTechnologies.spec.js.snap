// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`technology renders correctly 1`] = `
<section
  class="container_regular"
>
  <h2
    class="case_title_h2 m-24_bottom"
  >
    
    Technology stack
  
  </h2>
   
  <ul
    class="case_list-technologies m-96_bottom media-m-48_bottom"
  >
    <li
      class="case_technologies-item"
      tech-name="Golang"
    >
      <div
        class="case_technologies-item__tech-icon case_technologies-item__backend-go backend-go"
      />
       
      <p
        class="case_technologies-item__tech-name"
      >
        
    
  
      </p>
    </li>
    <li
      class="case_technologies-item"
      tech-name="Django"
    >
      <div
        class="case_technologies-item__tech-icon case_technologies-item__backend-django backend-django"
      />
       
      <p
        class="case_technologies-item__tech-name"
      >
        
    
  
      </p>
    </li>
    <li
      class="case_technologies-item"
      tech-name="PostgreSQL"
    >
      <div
        class="case_technologies-item__tech-icon case_technologies-item__backend-postgresql backend-postgresql"
      />
       
      <p
        class="case_technologies-item__tech-name"
      >
        
    
  
      </p>
    </li>
    <li
      class="case_technologies-item"
      tech-name="Nimble Streamer"
    >
      <div
        class="case_technologies-item__tech-icon case_technologies-item__other-nimble other-nimble"
      />
       
      <p
        class="case_technologies-item__tech-name"
      >
        
    
  
      </p>
    </li>
    <li
      class="case_technologies-item"
      tech-name="Plex Media Server"
    >
      <div
        class="case_technologies-item__tech-icon case_technologies-item__other-plex other-plex"
      />
       
      <p
        class="case_technologies-item__tech-name"
      >
        
    
  
      </p>
    </li>
    <li
      class="case_technologies-item"
      tech-name="jQuery"
    >
      <div
        class="case_technologies-item__tech-icon case_technologies-item__frontend-jquery frontend-jquery"
      />
       
      <p
        class="case_technologies-item__tech-name"
      >
        
    
  
      </p>
    </li>
    <li
      class="case_technologies-item"
      tech-name="Video.js"
    >
      <div
        class="case_technologies-item__tech-icon case_technologies-item__frontend-video-js frontend-video-js"
      />
       
      <p
        class="case_technologies-item__tech-name"
      >
        
    
  
      </p>
    </li>
    <li
      class="case_technologies-item"
      tech-name="Nginx"
    >
      <div
        class="case_technologies-item__tech-icon case_technologies-item__devops-nginx-txt devops-nginx-txt"
      />
       
      <p
        class="case_technologies-item__tech-name"
      >
        
    
  
      </p>
    </li>
  </ul>
</section>
`;
